scripts/exregional_get_verif_obs.sh

```diff
@@ -0,0 +1,456 @@
+# at the top of the hour, you should rename the file closest in time to
+# your hour(s) of interest to the above naming format. A script
+# "ush/mrms_pull_topofhour.py" is provided for this purpose. 
+#
+# If data is retrieved from HPSS, it will automatically staged by this
+# this script.
+#
+#
+# NDAS
+# ----------
+# If data is available on disk, it must be in the following 
+# directory structure and file name conventions expected by verification
+# tasks:
+#
+# {NDAS_OBS_DIR}/{YYYYMMDD}/prepbufr.ndas.{YYYYMMDDHH}
+# 
+# Note that data retrieved from HPSS and other sources may be in a
+# different format: nam.t{hh}z.prepbufr.tm{prevhour}.nr, where hh is 
+# either 00, 06, 12, or 18, and prevhour is the number of hours prior to
+# hh (00 through 05). If using custom staged data, you will have to
+# rename the files accordingly.
+# 
+# If data is retrieved from HPSS, it will automatically staged by this
+# this script.
+#
+#
+# NOHRSC
+# ----------
+# If data is available on disk, it must be in the following 
+# directory structure and file name conventions expected by verification
+# tasks:
+#
+# {NOHRSC_OBS_DIR}/{YYYYMMDD}/sfav2_CONUS_{AA}h_{YYYYMMDD}{HH}_grid184.grb2
+# 
+# where AA is the 2-digit accumulation duration in hours: 06 or 24
+#
+# METplus is configured to verify snowfall using 06- and 24-h accumulated
+# snowfall from 6- and 12-hourly NOHRSC files, respectively.
+#
+# If data is retrieved from HPSS, it will automatically staged by this
+# this script.
+
+#-----------------------------------------------------------------------
+# Create and enter top-level obs directory (so temporary data from HPSS won't collide with other tasks)
+mkdir -p ${OBS_DIR}
+cd ${OBS_DIR}
+
+# Set log file for retrieving obs
+logfile=retrieve_data.log
+
+# PDY and cyc are defined in rocoto XML...they are the yyyymmdd and hh for initial forecast hour respectively
+iyyyy=`echo ${PDY} | cut -c1-4`
+imm=`echo ${PDY} | cut -c5-6`
+idd=`echo ${PDY} | cut -c7-8`
+ihh=${cyc}
+
+# Unix date utility needs dates in yyyy-mm-dd hh:mm:ss format
+unix_init_DATE="${iyyyy}-${imm}-${idd} ${ihh}:00:00"
+
+# This awk expression gets the last item of the list $FHR
+fcst_length=`echo ${FHR}  | awk '{ print $NF }'`
+
+current_fcst=01
+while [[ ${current_fcst} -le ${fcst_length} ]]; do
+  #remove leading zero from current_fcst because bash treats numbers with leading zeros as octal *sigh*
+  current_fcst=$((10#${current_fcst}))
+  # Calculate valid date info using date utility  
+  vdate=`$DATE_UTIL -d "${unix_init_DATE} ${current_fcst} hours" +%Y%m%d%H`
+  unix_vdate=`$DATE_UTIL -d "${unix_init_DATE} ${current_fcst} hours" "+%Y-%m-%d %H:00:00"`
+  vyyyymmdd=`echo ${vdate} | cut -c1-8`
+  vhh=`echo ${vdate} | cut -c9-10`
+
+  # Calculate valid date + 1 day; this is needed because (for some ungodly reason) CCPA files for 19-23z
+  # are stored in the *next* day's 00h directory
+  vdate_p1=`$DATE_UTIL -d "${unix_init_DATE} ${current_fcst} hours 1 day" +%Y%m%d%H`
```

Comment:
Use more "modern" `$(...)` notation to enclose commands as opposed to backticks.